compute_endpoint/globus_compute_endpoint/engines/high_throughput/engine.py

```diff
@@ -300,7 +300,10 @@ def __init__(
         self._task_counter = 0
 
         try:
-            ipaddress.ip_address(address=address)
+            if address != "localhost":
+                # 'localhost'  works for both v4 and v6 in some circumstances
+                # where an actual numeric IP isn't required
+                ipaddress.ip_address(address=address)
```

Comment:
If `localhost` is allowed, then any name should be allowed.  This has historically required a valid IP address, which is coming from either the default (signature argument) or the caller of this `__init__` method, so I don't think this needs changing here.

Testing will confirm or rebuke the need.

Reply:
I'm not informed enough regarding how this address is used, other than the general 'local' communication need.  However, I did test (on Ubuntu) a few different parameters, after removing this check.  
1) localhost works
2) an arbitrary address obviously doesn't work:  ``zmq.error.ZMQError: Cannot assign requested address (addr='tcp://cnn.com:54419')``

so, to be safe and modify the current behavior as little as possible, I only make an exception for 'localhost' but do ip address validation otherwise.  If I remove the check altogether, that would need more testing of non-localhost DNS names, which I don't know how many lower layers can handle.

So, still leaning toward only making an exception for localhost to get rid of the dependence on 127.0.0.1.   Note that '::1' does not work when directly passed to zmq, even if it's a valid ipv6 address.

Reply:
`cnn.com` won't work because that's, well, not _this_ host (which I think your 2nd point is saying?).  But a friendly reminder that the [technical specification of ZMQ's socket .bind() method says](https://libzmq.readthedocs.io/en/zeromq3-x/zmq_tcp.html):

> When assigning a local address to a socket using _zmq_bind()_ with the _tcp_ transport, the _endpoint_ shall be interpreted as an _interface_ followed by a colon and the TCP port number to use.
>
> An interface may be specified by either of the following:
>
> * The wild-card `*`, meaning all available interfaces.
>
> * The primary IPv4 or IPv6 address assigned to the interface, in its numeric representation.
>
> * The non-portable interface name as defined by the operating system.

Meaning, it would, absent our ministrations, be completely valid to specify an interface -- indeed, that's the more fundamental detail from ZMQ's perspective!  We could handle `localhost` singularly, but I think we can be more general about this solve.

Reply:
But ... I think I've changed from my initial comment.  Meh.  This is a smaller detail than the need to make sure ipv6 is well handled.  Your call.

Reply:
I changed the validation to allow both hostname and ipv4/ipv6 addresses.  Added the utility method in class and in test, and if it's used elsewhere, we can move it to a more general location.